avocado/utils/pci.py

```
@@ -376,6 +376,96 @@ def get_vendor_id(full_pci_address):
+        ) from e
+
+
+def check_msix_capability(full_pci_address):
+    """
+    Check whether the PCI device supports Extended Message Signaled Interrupts
+    and if it is currently enabled.
+
+    :param full_pci_address: Full PCI address including domain (0000:03:00.0)
+    :return: True if supported, False otherwise
+    """
+    out = process.run(f"lspci -vvs {full_pci_address}", ignore_status=True, shell=True)
```

Ruff (0.14.0)

Function call with `shell=True` parameter identified, security issue

(S604)